clio-client/src/main/scala/org/broadinstitute/clio/client/commands/Commands.scala

```diff
@@ -2,39 +2,104 @@ package org.broadinstitute.clio.client.commands
 
 import akka.http.scaladsl.model.HttpResponse
 import com.typesafe.scalalogging.LazyLogging
+import io.circe.parser._
 import org.broadinstitute.clio.client.ClioClientConfig
+import org.broadinstitute.clio.client.parser.BaseArgs
+import org.broadinstitute.clio.client.util.IoUtil
+import org.broadinstitute.clio.client.webclient.ClientAutoDerivation._
+import org.broadinstitute.clio.client.webclient.ClioWebClient
+import org.broadinstitute.clio.transfer.model.{
+  TransferReadGroupV1Key,
+  TransferReadGroupV1Metadata,
+  TransferReadGroupV1QueryInput
+}
+import org.broadinstitute.clio.util.model.Location
 
 import scala.concurrent.{Await, ExecutionContext, Future}
 
 object Commands {
-  val addReadGroupBam = "addReadGroupBam"
-  val queryReadGroupBam = "queryReadGroupBam"
-}
+  sealed abstract class Command(
+    val commandName: String
+  )(implicit ec: ExecutionContext)
+      extends LazyLogging {
+    def execute(clioWebClient: ClioWebClient, config: BaseArgs): Boolean
 
-abstract class Command(name: String)(implicit ec: ExecutionContext)
-    extends LazyLogging {
+    def checkResponse(responseFuture: Future[HttpResponse]): Boolean = {
+      Await.result(
+        responseFuture.map[Boolean] { response =>
+          val isSuccess = response.status.isSuccess()
 
-  def execute: Boolean
+          if (isSuccess) {
+            logger.info(
+              s"Successfully completed command $commandName." +
+                s" Response code: ${response.status}"
+            )
+          } else {
+            logger.error(
+              s"Error executing command $commandName." +
+                s" Response code: ${response.status}"
+            )
+          }
+          isSuccess
+        },
+        ClioClientConfig.responseTimeout
+      )
+    }
+  }
+  case class AddReadGroupBam()(implicit ec: ExecutionContext)
+      extends Command(commandName = "addReadGroupBam") {
+    override def execute(clioWebClient: ClioWebClient,
+                         config: BaseArgs): Boolean = {
+      //parse metadata to validate inputs
+      val json = parse(IoUtil.readMetadata(config.metadataLocation)) match {
```

Comment:
I think it'd be more idiomatic to roll up `json` and `decoded` using `flatMap`:
```scala
val decodedOrErr = parse(...).flatMap(_.as[TransferReadGroupV1Metadata])
```
Then instead of throwing, we can transform the `Either` into a `Future`:
```scala
decodedOrErr.fold(Future.failed, { decoded => /* call to clioWebClient */ })
```
If we hit a failure, it'll still throw up at the `Await.result` layer.